File: bakery-inventory-backend/src/main/java/com/bakery/inventory/service/SupplierService.java
package com.bakery.inventory.service;

import com.bakery.inventory.dto.supplier.SupplierRequest;
import com.bakery.inventory.dto.supplier.SupplierResponse;

import java.util.List;

public interface SupplierService {

    SupplierResponse createSupplier(SupplierRequest request);

    List<SupplierResponse> getAllSuppliers();

    SupplierResponse getSupplierById(Integer id);

    SupplierResponse updateSupplier(Integer id, SupplierRequest request);

    void deactivateSupplier(Integer id);

    void activateSupplier(Integer id);
}

File: bakery-inventory-frontend/src/services/supplierService.js
import axiosInstance from '../api/axiosInstance';

/**
 * NEW FILE: SupplierService
 * Supplier directory and status management for inventory purchasing.
 */

export const supplierService = {
  getAllSuppliers: async () => {
    const response = await axiosInstance.get('/api/suppliers');
    return response.data;
  },

  getSupplierById: async (id) => {
    const response = await axiosInstance.get(`/api/suppliers/${id}`);
    return response.data;
  },

  createSupplier: async (supplierData) => {
    // CHANGE: Backend SupplierRequest expects { name, email, phone, address }
    const response = await axiosInstance.post('/api/suppliers', supplierData);
    return response.data;
  },

  updateSupplier: async (id, supplierData) => {
    const response = await axiosInstance.put(`/api/suppliers/${id}`, supplierData);
    return response.data;
  },

  activateSupplier: async (id) => {
    const response = await axiosInstance.patch(`/api/suppliers/${id}/activate`);
    return response.data;
  },

  deactivateSupplier: async (id) => {
    const response = await axiosInstance.patch(`/api/suppliers/${id}/deactivate`);
    return response.data;
  }
};
